lib/Dialect/Substrait/IR/Substrait.cpp

```diff
@@ -51,6 +51,27 @@ void SubstraitDialect::initialize() {
 namespace mlir {
 namespace substrait {
 
+LogicalResult
+CrossOp::inferReturnTypes(MLIRContext *context, std::optional<Location> loc,
+                          ValueRange operands, DictionaryAttr attributes,
+                          OpaqueProperties properties, RegionRange regions,
+                          llvm::SmallVectorImpl<Type> &inferredReturnTypes) {
+  Value leftInput = operands[0];
+  Value rightInput = operands[0];
+
+  TypeRange leftFieldTypes = leftInput.getType().cast<TupleType>().getTypes();
+  TypeRange rightFieldTypes = rightInput.getType().cast<TupleType>().getTypes();
+
+  llvm::SmallVector<mlir::Type> fieldTypes;
+  fieldTypes.append(leftFieldTypes.begin(), rightFieldTypes.end());
+  fieldTypes.append(rightFieldTypes.begin(), rightFieldTypes.end());
+  auto resultType = TupleType::get(context, fieldTypes);
+
+  inferredReturnTypes.emplace_back(resultType);
```

Comment:
Nit: You could also do

```
inferredReturnTypes = {resultType};
```

to make more explicit that its single return.

Reply:
It has to be

```cüü
inferredReturnTypes = SmallVector<Type>{resultType};
```

but I guess that's still more explicit for a single return?